sdk/appcenter-push/src/main/java/com/microsoft/appcenter/push/Push.java

```diff
@@ -79,6 +80,12 @@
      */
     private String mFirstGoogleMessageId;
 
+    /**
+     * Save Received message Intents to generate push notification
+     * when click notification from background
+     */
+    private HashSet<Intent> mReceivedIntents;
```

Comment:
Please use Map<String, Intent> where String is the googleMessageId, that way we can avoid the for loop.

Reply:
Please also add final and use interface and not implementation, also init it directly here with default constructor.

Reply:
Updated